aptos-move/framework/src/natives/cryptography/groups/arkworks_backend.rs

```diff
@@ -0,0 +1,1896 @@
+        }
+    }
+}
+
+fn element_mul_scalar_internal(
+    gas_params: &GasParameters,
+    context: &mut NativeContext,
+    ty_args: Vec<Type>,
+    mut args: VecDeque<Value>,
+) -> PartialVMResult<NativeResult> {
+    assert_eq!(1, ty_args.len());
+    let type_tag = context
+        .type_to_type_tag(ty_args.get(0).unwrap())?
+        .to_string();
+    let point_handle = pop_arg!(args, u64) as usize;
+    let scalar_handle = pop_arg!(args, u64) as usize;
+    match type_tag.as_str() {
+        "0x1::groups::BLS12_381_G1" => {
+            let point = context
+                .extensions()
+                .get::<Bls12381Context>()
+                .get_g1_point(point_handle);
+            let scalar = context
+                .extensions()
+                .get::<Bls12381Context>()
+                .get_scalar(scalar_handle);
+            let result = point.mul(scalar.into_repr());
+            let handle = context
+                .extensions_mut()
+                .get_mut::<Bls12381Context>()
+                .add_g1_point(result);
+            Ok(NativeResult::ok(
+                gas_params.bls12_381.g1_proj_mul * NumArgs::one(),
+                smallvec![Value::u64(handle as u64)],
+            ))
+        }
+        "0x1::groups::BLS12_381_G2" => {
+            let point = context
+                .extensions()
+                .get::<Bls12381Context>()
+                .get_g2_point(point_handle);
+            let scalar = context
+                .extensions()
+                .get::<Bls12381Context>()
+                .get_scalar(scalar_handle);
+            let result = point.mul(scalar.into_repr());
+            let handle = context
+                .extensions_mut()
+                .get_mut::<Bls12381Context>()
+                .add_g2_point(result);
+            Ok(NativeResult::ok(
+                gas_params.bls12_381.g2_proj_mul * NumArgs::one(),
+                smallvec![Value::u64(handle as u64)],
+            ))
+        }
+        "0x1::groups::BLS12_381_Gt" => {
+            let point = context
+                .extensions()
+                .get::<Bls12381Context>()
+                .get_gt_point(point_handle);
+            let scalar = context
+                .extensions()
+                .get::<Bls12381Context>()
+                .get_scalar(scalar_handle);
+            let result = point.pow(scalar.into_repr().as_ref());
+            let handle = context
+                .extensions_mut()
+                .get_mut::<Bls12381Context>()
+                .add_gt_point(result);
+            Ok(NativeResult::ok(
+                (gas_params.bls12_381.fr_to_repr + gas_params.bls12_381.fq12_pow_fr) * NumArgs::one(),
+                smallvec![Value::u64(handle as u64)],
+            ))
+        }
+        _ => {
+            Ok(NativeResult::err(InternalGas::zero(), E_UNKNOWN_GROUP))
+        }
+    }
+}
+
+fn ark_g1_affine_to_blst_g1_affine(ark: &ark_bls12_381::G1Affine) -> blst::blst_p1_affine {
```

Comment:
Are these ark to/from blst functions here for testing your assumptions about their different serialization formats?

Again, we cannot use `blst` with `unsafe`, unfortunately, so we'll have to move to `blstrs` instead.

But I think these should be moved into an appropriately-named file inside the `aptos-move/framework/tests` directory...